### DISCUSSION TITLE
typescript error when removing listeners from inside a disconnected event

### DISCUSSION THREAD

**Discussion by glend1** [djipco/webmidi#230] [Q&A]
as per the suggestion in #229, im posting this here in hopes of a solution. here is the code to reproduce;

````
WebMidi.addListener("disconnected", (e) => {
    e.port.removeListener()
})
````

and this is the full error returned `TS2349: This expression is not callable.
Each member of the union type '(<T extends keyof InputEventMap>(type?: Symbol | T, listener?: InputEventMap[T], options?: { channels?: number | number[]; context?: any; remaining?: number; }) => void) | (<T extends keyof OutputEventMap>(type?: Symbol | T, listener?: OutputEventMap[T], options?: { ...; }) => void)' has signatures, but none of those signatures are compatible with each other.`

any ideas?

**Comment by djipco** [comment]
Are there TypeScript users that can help with this. I'll gladly update the `.d.ts` file if someone can tell me what TypeScript expects in this case...

**Reply by glend1** [reply]
asking on something like stackoverflow might return a speedy resolution.

**Comment by djipco** [comment]
> asking on something like stackoverflow might return a speedy resolution.

Yes, probably. Unfortunately, I don't have much bandwidth to work on this right now. If you ever had the time, you could post the problem to StackOverflow and keep me updated. Otherwise, it will have to wait a little. It's probably something relatively simple but I'm lacking time...

**Comment by djipco** [comment]
I [posted this problem](https://stackoverflow.com/questions/70928770/typescript-union-type-with-incompatible-signatures) to StackOverflow. Hopefully, we'll get some help from TypeScript pros.

**Comment by djipco** [comment]
I implemented a workaround for this issue. See https://github.com/djipco/webmidi/issues/229#issuecomment-1038285328.
